packages/api-client-core/spec/GadgetRecord.spec.ts

```diff
@@ -442,4 +442,165 @@ describe("GadgetRecord", () => {
     product.setField("changed", true);
     expect(product.getField("changed")).toEqual(true);
   });
+
+  describe("record changes for current Date values against previous string value", () => {
+    it("correctly calculates hasChanges when the current value is the same Date as the previous value", () => {
+      const dateStr = "2024-04-19T21:03:37.000Z";
+
+      productBaseRecord = {
+        id: "123",
+        name: "A cool product",
+        body: "A description of why it's cool",
+        publishedAt: dateStr,
+      };
+
+      const product = new GadgetRecord<SampleBaseRecord>(productBaseRecord);
+      expect(product.changed()).toBe(false);
+
+      product.publishedAt = new Date(dateStr);
+      expect(product.changed()).toBe(false);
+    });
+
+    it("correctly calculates record changed when the current value is a Date and is the same date as the previous value", () => {
```

Comment:
maybe add that this test and the previous are different since one uses `changed` and the other is `changed("publishedAt")`. i was a bit confused since they looked almost identical